### DISCUSSION TITLE
Preparation for CPACS developer meeting

### DISCUSSION THREAD

**Discussion by MarAlder** [DLR-SL/CPACS#775] [General]
The next quarterly developer meeting will take place on **Mai 25, 2022**. 

I would like to try out this forum to prepare the discussions, allowing everyone to contribute with topics. 

# Background
The aim is to exchange on current CPACS  schema developments and to align with library developments, i.e.:

- Provide a platform for informal exchange on general topics
- Coordinate the development
- Identify responsibilities

If you want to join the meeting and contribute as a developer, please register [by mail](https://cpacs.de/pages/contact.html).

# CPACS Topics
In the last meeting, we decided not to prioritize a single issue, but to go over the main open issues in summary. In order to be able to prepare for these topics, I would like to give a short outline. Please add topics if you think we should discuss them as well.

## Interpretation and use of symmetry
Since we created a working group on this topic last time, I don't think we should discuss any details on it. The working group has not yet met. Therefore, we should clarify how we can push the topic further. A summary of the problem is given in [symmetry_cases.pptx](https://github.com/DLR-SL/CPACS/files/8574063/symmetry_cases.pptx).

## Parametric operations (using equations in CPACS)
Currently we link elements via `uID`s using only  `xsd:ID` and `xsd:IDREF` types. An additional level of parameterization would be given if we could not only link elements but also apply mathematical operations to them. This holds great opportunities and risks. We should decide whether this topic should be further pushed, and if yes - plan the next steps. 

@CLiersch, can you provide an XML example as comment so that people already think about pro's and con's?

## Usage of update element / provenance
It is currently not clear whether the last version number of a data set is the one in the `version` element or the one in the  `update` node. Please also make up your mind if the `update` node is an appropriate provenance solution or whether we should trigger different solutions.

```XML
    <header>
        <name>Basic Wing Model</name>
        <creator>Me</creator>
        <timestamp>2019-12-04T10:30:00</timestamp>
        <version>1.0</version>
        <cpacsVersion>3.3</cpacsVersion>
        <updates>
            <update>
                <cpacsVersion>3.4</cpacsVersion>
                <version>1.1</version>
                <modificatin>I updated to next CPACS release</modificatin>
            </update>
            <update>
                <cpacsVersion>3.3</cpacsVersion>
                <version>1.0</version>
                <modificatin>I created the dataset</modificatin>
            </update>
        </updates>
    </header>
```
or 
```XML
    <header>
        <name>Basic Wing Model</name>
        <creator>Me</creator>
        <timestamp>2019-12-04T10:30:00</timestamp>
        <version>1.1</version>
        <cpacsVersion>3.4</cpacsVersion>
        <updates>
            <update>
                <cpacsVersion>3.4</cpacsVersion>
                <version>1.1</version>
                <modificatin>I updated to next CPACS release</modificatin>
            </update>
           <update>
                <cpacsVersion>3.3</cpacsVersion>
                <version>1.0</version>
                <modificatin>I created the dataset</modificatin>
            </update>
        </updates>
    </header>
```

## Usage of toolspecifics
Please find out how the toolspecific node is understood and used in your environment. Please think of ideas how we can improve the situation. The correct approach is described in the [documentation](https://www.cpacs.de/documentation/CPACS_3_4_0_Docs/html/eb8cce31-1196-aad9-642f-59768e2366b7.htm) and [example file](https://github.com/DLR-SL/CPACS/blob/develop/examples/toolspecific.xml).

## CPACS examples and tutorials
Please think of ideas on how we can provide the community with better examples. Currently we have the [examples directory](https://github.com/DLR-SL/CPACS/tree/develop/examples). How can we support the use of CPACS, TiXI, TiGL, maybe RCE, etc.? Does someone has experience with creating [moocs](https://de.wikipedia.org/wiki/Massive_Open_Online_Course)? Should we set up a website with Jupyter Notebooks? Should we provide seminars? Let's collect ideas and initiate actions. 

# Software Libraries
I don't have much to report this time. How does it look from the TiGL side? Would you like to address certain topics, @joergbrech & @AntonReiswich?

**Comment by marcengelmann** [comment]
We at Bauhaus Luftfahrt would like to know if there are any activities regarding parametrization of hybrid propulsion components.

**Reply by MarAlder** [reply]
Good point. We'll talk about this in the meeting. An intermediate status from the EXACT project can be found in [CPACS_Powerbreakdown.pptx](https://github.com/DLR-SL/CPACS/files/8768495/CPACS_Powerbreakdown.pptx), but development continues.

**Comment by marcengelmann** [comment]
Regarding the toolspecific node we at Bauhaus Luftfahrt are curious to see other usages. We use this node heavily for our BLADE aircraft design environment where every tool in the chain has its own tool node.

**Comment by marcengelmann** [comment]
TIGL: Are there any activities regarding the visualization of additional components (e.g. landing gear) and what is the status of the CPACS creator development?
